### PR TITLE
Story 1-06

### PR DESCRIPTION
**Pull request by AnastasiaBlack**
Created services to provide the necessary methods for teacher's manipulations with groups. According to Story 1-06 the teacher can see the list of his groups. PS: role permissions still have to be implemented. Task is in progress.

### CODE REVIEW

File: src/main/java/com/softserve/teamproject/service/TeacherGroupsManipulationService.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import com.softserve.teamproject.entity.Group;
+import java.util.List;
+import org.springframework.security.access.prepost.PreAuthorize;
+
+public interface TeacherGroupsManipulationService {
+
+  /**
+   * The access to this method is secured and accessible only to the users with the "teacher" role.
+   * @param teachersName String value, name of the teacher
+   * @return List<Group>: </>List of Groups of this authenticated teacher
```
Comment (andrewkononenko) [inline src/main/java/com/softserve/teamproject/service/TeacherGroupsManipulationService.java]: Is it a typo "</>"?

Comment (AnastasiaBlack) [inline src/main/java/com/softserve/teamproject/service/TeacherGroupsManipulationService.java]: Oh..yes, it is! Thank you)

File: src/main/java/com/softserve/teamproject/service/TeacherGroupsManipulationService.java [hunk 2]
```diff
@@ -4,12 +4,17 @@
...
 public interface TeacherGroupsManipulationService {
 
   /**
    * The access to this method is secured and accessible only to the users with the "teacher" role.
+   *
    * @param teachersName String value, name of the teacher
-   * @return List<Group>: </>List of Groups of this authenticated teacher
+   * @return List<Group>: List of Groups of this authenticated teacher
    */
   @PreAuthorize("hasAuthority('teacher')")
```
Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/service/TeacherGroupsManipulationService.java]: @PreAuthorize("hasAuthority('teacher')") Story 6  => User with category role "It Academy" should have access to his groups. "It Academy" includes teacher, coordinator and admin.

File: src/main/java/com/softserve/teamproject/controller/TeacherGroupsController.java [hunk 3]
```diff
@@ -11,7 +11,7 @@
 import org.springframework.web.bind.annotation.RestController;
 
 /**
- * Controller for teacher's manipulations with the groups
+ * The controller provides methods for teacher's interaction with the groups.
  */
 @RestController
 public class TeacherGroupsController {
```
Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/controller/TeacherGroupsController.java]: not sure that we need different controller TeacherGroupsController for groups. GroupController would be better.

File: src/main/java/com/softserve/teamproject/repository/custom/GroupRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  @PersistenceContext
+  private EntityManager entityManager;
+
+  public List<Group> getAllGroupsOfTheTeacher(String teachersName) {
+    List<Group> allGroups = new ArrayList<>();
+    Iterable<Group> itGroups = entityManager
+        .createQuery("from Group group where group.teacher.nickName = :teachersName", Group.class)
+        .setParameter("teachersName", teachersName).getResultList();
+    itGroups.forEach(allGroups::add);
+    return allGroups;
```
Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/repository/custom/GroupRepositoryImpl.java]: What's the point to use such сonstructions if we can do this with QueryDslRepositorySupport in 2 lines. (1 line predicate + 1 line jpql query and no sql code ;D

**Pull request merged**